### PR TITLE
Fix file-server Makefiles

### PR DESCRIPTION
**Pull request by nicovank**
@dungwinux Please give this a test (`make measure` in each 5 directory), as before for some reason my laptop does not like file-server.

I would like to launch a set of measurements for this today, I know some benchmarks (Python...) take a while.

### CODE REVIEW

**Comment by dungwinux** [comment]
- [file-server] C++, Java, JS, Go work. Python for some reasons never detect the `initialized` line. Same behavior for Python/key-value
- I think the run-server-client.py script has a small issue where it does not kill the server after finished running the client.

File: JavaScript/file-server/file_server.js [hunk 1]
```diff
@@ -72,7 +72,8 @@ function main(args) {
 			handle_client(socket, line, args[2]);
 		});
 	}).listen(parseInt(args[1], 10), args[0]);
-	//}).listen(parseInt(args[0], 10), "127.0.0.1");
+
+	console.log("[info] initialized.");
```
Comment (dungwinux) [inline JavaScript/file-server/file_server.js]: Potentially a race condition. the log function should be put in the listen callback.
It should be something like this:

```js
.listen(parseInt(args[1], 10), args[0], () => { console.log("[info] initialized."); });
```

Comment (nicovank) [reply JavaScript/file-server/file_server.js]: Great, I thought `listen` was blocking, I will move to callback for this and `key-value`. Thanks!

**Comment by nicovank** [comment]
> Python for some reasons never detect the initialized line. Same behavior for Python/key-value

I think I need to run it in unbuffered mode. Will try this in a second.

> I think the run-server-client.py script has a small issue where it does not kill the server after finished running the client.

I actually think this is the OS not marking the port as free immediately since it's not closed. We could either add logic to catch signal and close port, or just wait a few seconds between runs. I want to try on the server first, maybe there's no issue there.

**Comment by nicovank** [comment]
Alright, this should work (minus the OS not marking port as available again, which is another issue). Can you give this one more try and let me know? Thanks.

**Comment by dungwinux** [comment]
The Python fix works.
Now that I test the `make measure`, it seems like the RAPL script messed it up.
Since we concat each parameters without quoting each, the parameter passing to `system` is different when it includes space, and thus I cannot run the measurement of run-server-client.py. Do you want to fix RAPL or the Makefile?

**Comment by nicovank** [comment]
The following I think would fix but I am stepping out for a bit and cannot test. I'm not the biggest fan since now if there are single quotes in command `'` this will cause issues. If you have a better fix please let me know, I will complete this later this evening. I think I rather not touch RAPL if possible.

```cpp
// OLD
std::string command = argv[2];
for (int i = 3; i < argc; ++i) {
    command.append(" ");
    command.append(argv[i]);
}

// NEW
std::string command = "'" + argv[2] + "'";
for (int i = 3; i < argc; ++i) {
    command.append(" ");
    command.append("'" + argv[i] + "'");
}
```

**Comment by dungwinux** [comment]
I think another  solution is to enclose another `'` outside `"`, so:

```
--server-command "./file_server.o 0.0.0.0 8080 ../../data/file-server"
```

would be

```
--server-command '"./file_server.o 0.0.0.0 8080 ../../data/file-server"'
```

**Comment by nicovank** [comment]
For run, on my laptop:
 -  key-value: All work but Python for me. Maybe concurrency problem? We should use 1 client job and reduce input size.
 -  file-server: For some reason still not working on my setup, any languages.

I will have to fix measure issue tomorrow. @dungwinux If you test your solution and it works please feel free to push to this branch. Thanks!

**Comment by dungwinux** [comment]
Seems like GitHub was bugging and I have to do an extra _empty_ commit to resolve a conflict.
RAPL script should work nicely after the _add extra quotations_ commit.

**Comment by dungwinux** [comment]
Both key-value and file-server seems to quit properly now.

**Comment by nicovank** [comment]
Ok, this should finally work (works for me on server). Give this one last look / test then let's merge. Launching preliminary run on server to see if it works all good by meeting today.

**Pull request merged**